internal/datacoord/meta.go

```diff
@@ -334,6 +335,136 @@ func (m *meta) UpdateFlushSegmentsInfo(
 	return nil
 }
 
+// UpdateDropChannelSegmentInfo updates segment checkpoints and binlogs before drop
+// reusing segment info to pass segment id, binlogs, statslog, deltalog, start position and checkpoint
+func (m *meta) UpdateDropChannelSegmentInfo(channel string, segments []*SegmentInfo) error {
```

Comment:
This function is too long, we can split it to multi functions to improve readability.

Reply:
Fixed